actions/upload_sessions.go

```diff
@@ -367,22 +370,37 @@ func sqlWhereForGenesisHashAndChunkIdx(genesisHash string, chunkIdx int) string
 	return fmt.Sprintf("(genesis_hash = '%s' AND chunk_idx = %d)", genesisHash, chunkIdx)
 }
 
-func waitForTransfer(ethAddr string, uploadSessionId string, isAlpha bool) {
-	success, _ := services.EthWrapper.WaitForTransfer(services.StringToAddress(ethAddr))
+func waitForTransferAndNotifyBeta(alphaEthAddr string, betaEthAddr string, uploadSessionId string, isAlpha bool) {
+	transferAddr := services.StringToAddress(alphaEthAddr)
+	balance, err := services.EthWrapper.WaitForTransfer(transferAddr)
+
+	paymentStatus := models.PaymentStatusConfirmed
+	if err != nil {
+		paymentStatus = models.PaymentStatusError
+	}
 
 	session := models.UploadSession{}
 	if err := models.DB.Find(&session, uploadSessionId); err != nil {
+		raven.CaptureError(err, nil)
 		return
 	}
 
-	if success {
-		session.PaymentStatus = models.PaymentStatusConfirmed
-	} else {
-		session.PaymentStatus = models.PaymentStatusError
-	}
-
+	session.PaymentStatus = paymentStatus
 	if err := models.DB.Save(&session); err != nil {
+		raven.CaptureError(err, nil)
 		return
 	}
-	// TODO(pzhao5): do other stuff here
+
+	// Alpha send half of it to Beta
+	if isAlpha && paymentStatus == models.PaymentStatusConfirmed {
```

Comment:
Instead of passing in `isAlpha` can we use the session type from the uploadSession struct to determine if this is the alpha or beta broker? It would reduce the responsibility of the caller to know whether to pass true of false.